### ISSUE TITLE
ANNOTATION_DEF token support in RightCurlyCheck

### ISSUE THREAD

**Issue by sd1998** [checkstyle/checkstyle#7041]
Currently [RightCurly](https://checkstyle.sourceforge.io/config_blocks.html#RightCurly) accepts a lot less tokens (13 tokens) than [LeftCurly](https://checkstyle.sourceforge.io/config_blocks.html#LeftCurly) (20 tokens).
It is unclear if this is just a mistake or if there is some underlying reason. This issue aims at adding support for `ANNOTATION_DEF` token in `RightCurlyCheck`. 

Solving this issue will help solve Issue #3547.

````
$ cat TestClass.java
public @interface TestAnnotation {
          int someValue();} // expected violation

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="RightCurly">
  <property name="option" value="alone"/>
</module>
    </module>
</module>

$ java -jar checkstyle-8.24-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

Expected a violation for the right curly brace or a way to enable violation on the token.

**Comment by rnveach** [comment]
Fix was merged.
